
```{r error=TRUE}
data.geno<- read.csv("./RiceSNPData/Rice_44K_genotypes.csv.gz", row.names=1, na.strings=c("NA","00")) #import csv data of SNPs
```

```{r error=TRUE}
# Structure and summary of the first 20 columns of data.geno
head(data.geno [,1:20])
summary(data.geno[,1:20])
```

```{r error=TRUE}
data.geno.2500 <-data.geno[,sample(ncol(data.geno),2500)] # Random selection of 2500 SNPs
dim(data.geno.2500)
```


## Creating a MDS plot

```{r error=TRUE}
geno.numeric<-data.matrix(data.geno.2500) #convert data matrix to numbers
head(geno.numeric[,1:20])

genDist<-as.matrix(dist(geno.numeric)) # calculate Euclidean distance between each rice variety
geno.mds<-as.data.frame(cmdscale(genDist)) #multi-dimesnional scaling
head(geno.mds)
```

- geno.mds contains genotypic information rescaled to display in two dimesnions

### Plot 

```{r error=TRUE}
library(ggplot2)
pl <- ggplot(data=geno.mds, mapping=aes(x=V1, y=V2))
pl<-pl+geom_point()
pl
```

### Merge MDS scaled genotype data with the phenotype data by "row.name" column

```{r error=TRUE}
data.pheno<- read.csv("./RiceSNPData/RiceDiversity.44K.MSU6.Phenotypes.csv", row.names=1, na.strings=c("NA","00"))
head(data.pheno[,1:20])
summary(data.pheno[,1:20])

genopheno<-merge(geno.mds,data.pheno, by="row.names", all=T) #merged genotype and phenotype data
head(genopheno)
summary(genopheno)
```

### Plots to see if subgroups vary by:

```{r error=TRUE}
# a) AMYLOSE CONTENT 

pl<-ggplot(data=genopheno,mapping=aes(x=V1,y=V2,color=Amylose.content))+scale_color_gradientn(colors=topo.colors(25))
pl<-pl+geom_point()
pl

# b) PERICARP COLOR
pl<-ggplot(data=genopheno,mapping=aes(x=V1,y=V2,color=Pericarp.color))
pl<-pl+geom_point()
pl

#c) REGION
pl<-ggplot(data=genopheno,mapping=aes(x=V1,y=V2,color=Region))
pl<-pl+geom_point()
pl
```

# Create the bim file 

- The bim file provides information on the alleles

```{r error=TRUE}
bim <- data.geno.2500.fs[,-1:-6]

colnames(bim) <- colnames(data.geno.2500)

bim[bim=="-9"] <- NA

bim <- apply(bim,2,function(x) unique(na.omit(x))) 

bim[,1:5]
```

### Seperate SNP_id to get chromosome and position 

```{r error=TRUE}
bim <- bim %>% 
  separate(SNP_ID,into = c("chromosome","position"),sep="_",remove=FALSE) %>% # create a column for chromosome and position
  select(chromosome, SNP_ID, cM, position, allele1=V1, allele2=V2) # get columns in right order

head(bim)
```

## STATISTICAL ANALYSIS: Obtain the mean of the trait for each population

```{r error=TRUE}
mean(data.pheno.mds$Protein.content,na.rm=T) ##the na.rm argument tells R to ignore missing data coded by "NA"

tapply(X=data.pheno.mds$Protein.content,INDEX=data.pheno.mds$popID,FUN=min,na.rm=T) #The first argument to tapply() is the data we want to summarize, the second argument is how the data should be split, the third argument is the function we want to apply (min in this case) and the fourth argument are additional arguments to be passed to the function listed in the argument.
```

### ANOVA

```{r error=TRUE}
aov1 <- aov(Protein.content ~ popID,data=data.pheno.mds) #1-way ANOVA for Protein.content by population
summary(aov1)
```

# GWAS Analysis

```{r error=TRUE}
#load information about snp chromosome and position
snpinfo <- read.csv("~/Assignment_4_Singh.Mitali/RiceSNPData/snpInfo.csv",row.names=1)
head(snpinfo) #note one column for chromosome and one for position (in base pairs)
data.geno <- read.csv("/home/ubuntu/Assignment_4_Singh.Mitali/RiceSNPData/Rice_44K_genotypes.csv.gz", row.names=1, na.strings=c("NA","00")) 
```

### Merge genotypic and phenotypic data
```{r error=TRUE}

head(data.pheno.mds) 
rownames(data.pheno.mds) <- data.pheno.mds$Row.names
data.geno.pheno <- merge(data.pheno.mds,data.geno,by="row.names") # merged data
```

### Using SNPassoc
```{r error=TRUE}
library(SNPassoc)
data.geno.pheno3 <- data.geno.pheno[,c(1:47,grep("X3_",colnames(data.geno.pheno)))]
snpinfo3 <- snpinfo[snpinfo$chr==3,] #create new data frames containing only chromosome 3 information.
#grep() is the R version of the linux grep command that you saw in Ian's section.  So the command below extracts columns 1:47 and all chromosome 3 genotype columns

#convert SNPinfo to a format that SNPassoc can use
snps3 <- setupSNP(data.geno.pheno3,48:ncol(data.geno.pheno3),sort=T,info=snpinfo3,sep="")
```

### GWAS Analysis without Population structure correction
```{r error=TRUE}
wg3 <- WGassociation(Alu.Tol,data=snps3,model="co",genotypingRate=50)
plot(wg3,print.label.SNPs=FALSE)
```

### Comparing expected and observed SNP association in the dataset
```{r error=TRUE}
qqpval(codominant(wg3)) # quantile-quantile plot: codominant(wg3) returns the observed p-values.

```

### GWAS analysis with Population Correction
```{r error=TRUE}
#analysis with population structure correction:
wg3.corrected <- WGassociation(Alu.Tol ~ pop1 + pop2 + pop3 + pop4,data=snps3,model="co",genotypingRate=50)
#analysis with population structure correction.
plot(wg3.corrected,print.label.SNPs=FALSE)
sum(codominant(wg3.corrected) < 1e-5)

qqpval(codominant(wg3.corrected))

```

- There is a significant amount of difference in the output before and after population correction. There were a lot more SNPs before population correction (false positives)